packages/lab/src/overlay/Overlay.tsx

```diff
@@ -40,14 +30,13 @@ export const Overlay = forwardRef<HTMLDivElement, OverlayProps>(
       onOpenChange,
       placement: placementProp = "top",
       id: idProp,
-      onClose,
       ...rest
     } = props;
 
     const id = useId(idProp);
     const arrowRef = useRef<SVGSVGElement | null>(null);
 
-    const [openState, setOpenState] = useControlled({
+    const [openState, setOpenState, isControlled] = useControlled({
```

Comment:
This seems unnecessary and doesn't match how we've done this elsewhere. We should create a handler in the code that sets the state and calls onOpenChange. Uncontrolled components should still call the event handlers.